### PR TITLE
Host key "authorize" cleanup

### PR DESCRIPTION
**Pull request by stefwalter**
Various bits of cleanup for the "authorize" message about asking for a host key.

### CODE REVIEW

File: src/ssh/cockpitsshrelay.c [hunk 1]
```diff
@@ -306,7 +306,7 @@ challenge_for_knownhosts_data (CockpitSshData *data)
   gchar *ret = NULL;
   gchar *response = NULL;
 
-  response = challenge_for_auth_data ("host-key", data->outfd, NULL);
+  response = challenge_for_auth_data ("x-host-key", data->outfd, NULL);
```
Comment (dperpeet) [inline src/ssh/cockpitsshrelay.c]: do we need to change the setting [here](https://github.com/stefwalter/cockpit/blob/e1f3a763e6e947f6a63b4e39198c1f3d8a0d0a55/src/ssh/cockpitsshrelay.c#L1169) as well?

Comment (stefwalter) [inline src/ssh/cockpitsshrelay.c]: No that's not related.

**Comment by dperpeet** [comment]
Is this related?
```
# testBasic (check_dashboard.TestBasicDashboard)
#
Unexpected journal message 'cockpit-ssh-Message: received unexpected "authorize" control message'
Unexpected journal message 'cockpit-ssh-Message: received unexpected "authorize" control message'
Unexpected journal message '/usr/libexec/cockpit-pcp: bridge was killed: 11'
```

**Comment by dperpeet** [comment]
remaining failures don't seem related

**Pull request closed without merging**